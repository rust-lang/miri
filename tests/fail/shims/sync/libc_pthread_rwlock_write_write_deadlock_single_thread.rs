//@ignore-target-windows: No libc on Windows

fn main() {
    let rw = std::cell::UnsafeCell::new(libc::PTHREAD_RWLOCK_INITIALIZER);
    unsafe {
        assert_eq!(libc::pthread_rwlock_wrlock(rw.get()), 0);
        libc::pthread_rwlock_wrlock(rw.get()); //~ ERROR: deadlock
    }
}
