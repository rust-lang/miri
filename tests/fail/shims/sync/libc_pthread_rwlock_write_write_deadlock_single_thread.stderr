error: deadlock: the evaluated program deadlocked
  --> $DIR/libc_pthread_rwlock_write_write_deadlock_single_thread.rs:LL:CC
   |
LL |         libc::pthread_rwlock_wrlock(rw.get());
   |                                             ^ the evaluated program deadlocked
   |
   = note: inside `main` at $DIR/libc_pthread_rwlock_write_write_deadlock_single_thread.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

